test/v1/server/integration_test.go

```diff
@@ -19,45 +19,169 @@ package server
 )
 
-func SetupSuites() []suite.TestingSuite {
-	var suites []suite.TestingSuite
-	suites = append(suites, &CollectionSuite{
-		database: fmt.Sprintf("integration_db1_%x", rand.Uint64()),
-	})
+var (
+	testDB         string
+	testCollection = "test_collection"
+)
+
+type Map map[string]interface{}
+type Doc Map
+
+var testCreateSchema = map[string]interface{}{
+	"schema": map[string]interface{}{
+		"title":       testCollection,
+		"description": "this schema is for integration tests",
+		"properties": map[string]interface{}{
+			"pkey_int": map[string]interface{}{
+				"description": "primary key field",
+				"type":        "integer",
+			},
+			"int_value": map[string]interface{}{
+				"description": "simple int field",
+				"type":        "integer",
+			},
+			"string_value": map[string]interface{}{
+				"description": "simple string field",
+				"type":        "string",
+				"maxLength":   128,
+			},
+			"added_string_value": map[string]interface{}{
+				"description": "simple string field",
+				"type":        "string",
+			},
+			"bool_value": map[string]interface{}{
+				"description": "simple boolean field",
+				"type":        "boolean",
+			},
+			"double_value": map[string]interface{}{
+				"description": "simple double field",
+				"type":        "number",
+			},
+			"added_value_double": map[string]interface{}{
+				"description": "simple double field",
+				"type":        "number",
+			},
+			"bytes_value": map[string]interface{}{
+				"description": "simple bytes field",
+				"type":        "string",
+				"format":      "byte",
+			},
+			"uuid_value": map[string]interface{}{
+				"description": "uuid field",
+				"type":        "string",
+				"format":      "uuid",
+			},
+			"date_time_value": map[string]interface{}{
+				"description": "date time field",
+				"type":        "string",
+				"format":      "date-time",
+			},
+			"array_value": map[string]interface{}{
+				"description": "array field",
+				"type":        "array",
+				"items": map[string]interface{}{
+					"type": "object",
+					"properties": map[string]interface{}{
+						"id": map[string]interface{}{
+							"type": "integer",
+						},
+						"product": map[string]interface{}{
+							"type": "string",
+						},
+					},
+				},
+			},
+			"object_value": map[string]interface{}{
+				"description": "object field",
+				"type":        "object",
+				"properties": map[string]interface{}{
+					"name": map[string]interface{}{
+						"type": "string",
+					},
+				},
+			},
+		},
+		"primary_key": []interface{}{"pkey_int"},
+	},
+}
+
+func setupTests(t *testing.T) (string, string) {
+	db := fmt.Sprintf("integration_%s_%x", t.Name(), rand.Uint64())
+	//	coll := fmt.Sprintf("integration_coll1_%x", rand.Uint64())
+	dropDatabase(t, db)
+	createDatabase(t, db).Status(http.StatusOK)
+	createCollection(t, db, testCollection, testCreateSchema).Status(http.StatusOK)
+
+	return db, testCollection
+}
+
+func cleanupTests(t *testing.T, db string) {
```

Comment:
nit: instead of this you can return the cleanup func from the setupTests so that whatever we are setting up the cleanup is in sync with that setup.

Reply:
There is not much difference. setup and cleanup tests defined next to each other.
But yes, we may change it in the future.

Reply:
yes, but this way of setup and cleanup is very broad and maybe not be applicable for all use-cases. Attaching it to the setup reduces it.